src/app.ts

```diff
@@ -13,14 +14,47 @@ interface TemperatureSummary {
   average: number
 }
 
+const readValues: TemperatureReading[] = []
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  readValues.push(...readings)
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
+  const summary: TemperatureSummary = {
+    first: 0,
+    last: 0,
+    high: 0,
+    low: 0,
+    average: 0,
+  }
+
+  const filteredValues = readValues.filter(
+    (value) =>
+      value.time.getTime() === new Date(date).getTime() && value.city === city,
+  )
+
+  const filteredValuesLength: number = filteredValues.length
+  const temperatureArray: number[] = []
+
+  if (filteredValuesLength > 0) {
+    summary.first = filteredValues[0].temperature
+    summary.last = filteredValues[filteredValuesLength - 1].temperature
+
+    for (let i = 0; i < filteredValues.length; i++) {
+      temperatureArray.push(filteredValues[i].temperature)
+    }
+
+    summary.high = Math.max(...temperatureArray)
+    summary.low = Math.min(...temperatureArray)
+    summary.average =
+      temperatureArray.reduce((a, b) => a + b, 0) / filteredValuesLength
+
+    return summary
+  } else {
```

Comment:
you can remove the `else` condition and only leave `return null`

Reply:
you already have a `return` inside of the `if`, so if the condition is fulfilled, the code will never reach the `return null`